### PR TITLE
add architecture docs

### PR DESCRIPTION
**Pull request by csantanapr**

### CODE REVIEW

File: docs/arch/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+### Create Trigger Feed
+![MessageHub Trigger Create](images/Arch-Provider-MHV1-Create.png)
+
+**Scenario:** User wants to create a trigger `trigger1` for MessageHub service instance `instance1`, using Credentials `Credential-1` and use rule `rule1` to invoke action `action1` with messages from topic `topic1`.
+
+1. Developer creates a MessageHub service `instance1`
+2. Developer creates topic `topic1` in MessageHub service `instance1`
+3. Developer creates Credential key `Credential-1` for MessageHub `instance1`
+4. Developer creates trigger `trigger1` on OpenWhisk, the trigger stores the annotation `feed` with the feedAction name from system package or binded package.(`/whisk.system/messagingWeb/messageHubFeed`).
+5. Developer invokes action feedAction to create trigger feed passing input parameters (lifeCycle:`CREATE`, trigger1, Credentials1, Options:`topic1`)
```
Comment (daisy-ycguo) [inline docs/arch/README.md]: Who is `developer` here: "**Developer** invokes action feedAction" ? May need to clarify.

File: docs/arch/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+6. The feedAction invokes feedWebAction forwarding input parameter.
+7. The feedWebAction inserts trigger feed doc into DB for worker group 0 (feedWebActions protects DB credentials)
```
Comment (daisy-ycguo) [inline docs/arch/README.md]: use singular format: feedWebActions => feedWebAction

File: docs/arch/README.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+8. DB insertion notifies workers group 0 via Cloudant/CoucDB changes API, workers listen on DB view with a filter for their group `worker0` and gets the DB doc.
```
Comment (daisy-ycguo) [inline docs/arch/README.md]: typo: Cloudant/CoucDB => Cloudant/CouchDB

File: docs/arch/README.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+9. Kafka Consumer is created on each worker in a consumer group and starts polling for messages on `topic1` from `instance1` using `Credentials-1`.
+10. Developer creates rule1 indicating that when trigger1 fires invoker action1
```
Comment (daisy-ycguo) [inline docs/arch/README.md]: Add backtick symbol: Developer creates `rule1` indicating that when `trigger1` fires invoker `action1`

File: docs/arch/README.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+11. Event source produces messages on `topic2`
```
Comment (daisy-ycguo) [inline docs/arch/README.md]: typo: `topic2` => `topic1`

File: docs/arch/README.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+3. Developer creates Credential key `Credential-1` for MessageHub `instance1`
+4. Developer creates trigger `trigger1` on OpenWhisk, the trigger stores the annotation `feed` with the feedAction name from system package or binded package.(`/whisk.system/messagingWeb/messageHubFeed`).
+5. Developer invokes action feedAction to create trigger feed passing input parameters (lifeCycle:`CREATE`, trigger1, Credentials1, Options:`topic1`)
+6. The feedAction invokes feedWebAction forwarding input parameter.
+7. The feedWebAction inserts trigger feed doc into DB for worker group 0 (feedWebActions protects DB credentials)
+8. DB insertion notifies workers group 0 via Cloudant/CoucDB changes API, workers listen on DB view with a filter for their group `worker0` and gets the DB doc.
+9. Kafka Consumer is created on each worker in a consumer group and starts polling for messages on `topic1` from `instance1` using `Credentials-1`.
+10. Developer creates rule1 indicating that when trigger1 fires invoker action1
+11. Event source produces messages on `topic2` 
+12. Both consumers will batch the messages from topic1 and fire trigger1.
```
Comment (daisy-ycguo) [inline docs/arch/README.md]: Add backtick symbol: from `topic1` and fire `trigger1`.

File: docs/arch/README.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    - The fire is done with an http request containing the batch of messages in the body.
+    - Consumer will not poll for more messages and will not commit batch of messages until the http request gets a response from OpenWhisk trigger endpoint.
+    - Consumers in the same consumer group get assigned a set of partitions, each consumer on each woker host will get a unique set of messages avoiding duplicate messages being included in trigger fires.
```
Comment (daisy-ycguo) [inline docs/arch/README.md]: typo: woker => worker

**Comment by rabbah** [comment]
Fixed all the typos and addressed review comments.

**Comment by csantanapr** [comment]
Wow I forgot about this PR. Thanks @rabbah

**Pull request merged**